flink-clients/src/test/java/org/apache/flink/client/deployment/application/ApplicationDispatcherBootstrapTest.java

```diff
@@ -348,6 +350,13 @@ void testApplicationIsStoppedWhenStoppingBootstrap() throws Exception {
 
         bootstrap.stop();
 
+        // EmbeddedExecutor calls getJobStatus after the job is submitted in a busy-waiting loop to
+        // wait for the job to pass the initialization phase. Only then, a JobClient is returned
+        // which finalizes the job submission. Completing the getJobStatusFuture after calling
+        // ApplicationDispatcherBootstrap#stop ensures that the applicationExecutionFuture doesn't
+        // complete before ApplicationDispatcherBootstrap#stop is called
+        getJobStatusFuture.complete(JobStatus.RUNNING);
```

Comment:
This seems to rely on very particular behavior of the embedded executor.

Can we achieve the same by delaying the acknowledgement of the job submission? That'd seem more intuitive to me.

Reply:
You have a point. I went through the process once more and found a solution that's even closer to the `ApplicationDispatcherBootstrap` implementation. We can utilize the `ManuallyTriggeredScheduledExecutor` to trigger the bootstrap execution task.